File: ios/Kwift/com/lightningkite/kwift/observables/EventToObservableProperty.actual.swift
import RxSwift
import RxRelay

//--- Observable<Box<T>>.asObservablePropertyUnboxed(T)
extension Observable where Observable.Element: BoxProtocol {
    public typealias T = Observable.Element.T
    public func asObservablePropertyNullable(defaultValue: T) -> ObservableProperty<T> {
        return EventToObservableProperty<T>(defaultValue, self.map { it in it.asBox() })
    }
    public func asObservablePropertyNullable(_ defaultValue: T) -> ObservableProperty<T> {
        return asObservablePropertyNullable(defaultValue: defaultValue)
    }
}

File: android/src/main/java/com/lightningkite/khrysalis/observables/EventToObservableProperty.actual.kt
package com.lightningkite.khrysalis.observables

import com.lightningkite.khrysalis.Box
import io.reactivex.Observable


fun <T> Observable<Box<T>>.asObservablePropertyUnboxed(defaultValue: T): ObservableProperty<T> {
    return EventToObservableProperty<T>(defaultValue, this)
}
